src/types/job.rs

```diff
@@ -48,8 +47,7 @@ pub struct JobDescriptor {
 )]
 pub struct SubmitPackageRequest {
     /// The 'type' of package, NPM, RubyGem, etc
-    // don't deprecate until `ecosystems` is live.
-    //#[deprecated = "No longer used."]
+    #[deprecated = "No longer used."]
     #[serde(rename = "type")]
     pub package_type: PackageType,
```

Comment:
With this change, we get the following warning in the CLI:

```
warning: use of deprecated field `phylum_types::types::job::SubmitPackageRequest::package_type`: No longer used.
   --> cli/src/api/mod.rs:305:13
    |
305 |             package_type: req_type.to_owned(),
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(deprecated)]` on by default
```

You can see the code [here](https://github.com/phylum-dev/cli/blob/2fd832d6588d20ea88a8af8c7519216752ec43e5/cli/src/api/mod.rs#L295-L316).

My problem is this: We have to construct a `SubmitPackageRequest` in order to use the API. We can't construct it without this field. So we are now forced to put `#[allow(deprecated)]` on that construction... (And as far as I can tell, we have to put it on the entire construction. Putting it on just the offending field does not work).

Is that really desired? Am I missing something?

Reply:
I think that's how it is, unfortunately. In API we have `#[allow(deprecated)]` where we create structs containing deprecated fields to return them.

We might also be able to just get rid of the fields since their replacements have been around for months now.

Reply:
> I think that's how it is, unfortunately. In API we have `#[allow(deprecated)]` where we create structs containing deprecated fields to return them.

I guess this is just another downside to the shared types. Ideally, we would just deprecate these fields on the consumer side and not the producer side to avoid this. We could probably do that with feature flags (something like `#[cfg_attr(feature = "server", deprecated)]`)... But that's probably not worth it.

> We might also be able to just get rid of the fields since their replacements have been around for months now.

This has the same problem as above because it's a shared type. We need to remove a value from the consumer side before we can remove it from the producer side. So for `package_type`, we need to remove it from API **before** we remove it from CLI. But since the types are shared, that's complicated.